clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -494,7 +505,8 @@ private void handleShareAcknowledgeSuccess(Node fetchTarget,
             }
 
             final short requestVersion = resp.requestHeader().apiVersion();
-            if (acknowledgeRequestState.isCommitSync() && !handler.handleResponse(response, requestVersion)) {
+
+            if (!handler.handleResponse(response, requestVersion)) {
```

Comment:
Here, can we keep the `isCommitSync()` check as the second condition, that would ensure the `handleResponse` is called, epochs are updated and then the commitSync check is done.

Reply:
We'll handle this as a follow-on task. I think it might be a bit more subtle.